### DISCUSSION TITLE
KlipperScreen on one Pi for 2 printers ignores my port definition and default printer definition

### DISCUSSION THREAD

**Discussion by WernherStoltz** [KlipperScreen/KlipperScreen#1046] [Q&A] [answered]
I have searched all the forums and I feel that maybe I need to put this under a new discussion title as all the ones I have seen doesn't address my question and some have been left open without responses for months while others have been closed and I still can't figure out from the responses how to get the issue resolved.

I have done everything to work through all the instructions and looked at every issue that was logged about this issue but I cannot get it to work correctly.

Here is what I am trying to do:

Run KlipperScreen on 1 Pi to control 2 printers.

This is my configuration:
In my /home/pi folder I have a KlipperScreen.Conf file: (Note the default printer which is ignored)

~~~~~~~~
[include FW401_data/config/KlipperScreen.conf]

[include FW103_data/config/KlipperScreen.conf]

# If multiple printers are defined, this can be set the name of the one to show at startup.
default_printer: Ender 3 S1 Pro FW401

#~# --- Do not edit below this line. This section is auto generated --- #~#
#~#
#~# [main]
#~# screen_blanking = off
#~#
;
The first two statements refers to the location of the KlipperScreen.conf files for the 2 printers and they contain the printer definitions:

/home/wernher/FW103_data/config/KlipperScreen.conf: (Note port 7125 which is the default and is correct for this printer)

~~~~~~~~
[printer Ender 3 S1 Pro FW103] moonraker_host: 127.0.0.1 moonraker_port: 7125 moonraker_api_key: False

and

/home/wernher/FW401_data/config/KlipperScreen.conf: (Note port 7126 which is changed to the default and is incorrect for this printer)

~~~~~~~~
[printer Ender 3 S1 Pro FW401 moonraker_host: 127.0.0.1 moonraker_port: 7126 moonraker_api_key: False

Here is the log file:

[KlipperScreen.log](https://github.com/KlipperScreen/KlipperScreen/files/12041586/KlipperScreen.log)

Based on what I see in the log the printers are correctly identified and parsed but when I get to my KlipperScreen it asks me to select a printer and correctly shows my two printers. When I select them, it goes to the correct printer but any instructions that I issue only goes to my first printer. If I look at the printer definitions in KlipperScreen, they are both showing port 7125 which is the default port.

I need a very thorough explanation of why this is happening as everywhere I look everybody just says it is so easy, just add the printers in the conf file, but the reality it is not working that way.

I am at the point now where I'm happy to just proceed without it as teh amount of effot to get it working is just not worth it anynore.

**Comment by alfrix** [comment]
> Based on what I see in the log the printers are correctly identified and parsed

no they are not only the header is parsed, attach one or both config files (do not copy just attach zipped)
FW401_data/config/KlipperScreen.conf
FW103_data/config/KlipperScreen.conf

i suspect it has to do with missing line breaks, because it has happened multiple times, idk if people do this intentionally or there is a program responsible, but i want to see the actual files because to try and detect and give a big error in the interface

**Comment by WernherStoltz** [comment]
Thanks @alfrix 

I've added the two conf files under FW103 and FW401

[KlipperScreen.zip](https://github.com/KlipperScreen/KlipperScreen/files/12042974/KlipperScreen.zip)

Maybe I should try and edit the config files with Notepad++ and upload them in UTF-8 format then, but I'll wait for your response.

**Comment by alfrix** [comment]
as suspected no newlines in the files provided,
each configuration should go on a separate line

![imagen](https://github.com/KlipperScreen/KlipperScreen/assets/1247237/09e6a6ad-2020-4457-874e-5ba046fcc1ad)

did you do this on purpose, how did you create those files?

**Reply by WernherStoltz** [reply]
I did that based on an example. Initially I had it as separate lines in the top KlipperScreen.conf where it didn't work until I moved it down to the printer_data/config folder.

I'll quickly try that and see i fit sorts the problem out.

**Comment by WernherStoltz** [comment, marked as answer]
@alfrix you ae my hero 🤪

It works like a charm now. Why is it not accepting the default printer though?

In the /home/pi folder I created the includes to the two printer_data/config folders where the individual KlipperScreen.conf files are, but just below the include lines I have the default printer defined.

This is what it looks like:

[include FW401_data/config/KlipperScreen.conf]

[include FW103_data/config/KlipperScreen.conf]

# If multiple printers are defined, this can be set the name of the one to show at startup.

default_printer: Ender 3 S1 Pro FW401

#~# --- Do not edit below this line. This section is auto generated --- #~#
#~#
#~# [main]
#~# screen_blanking = off
#~#

And this time there are line feeds 😂

**Comment by WernherStoltz** [comment]
@alfrix 
I am happy with what we achieved with your directions and will close this discussion based on that. The default printer is still a mystery but one that I can live with.

Thanks for you great help. It is much appreciated.

**Reply by tkunchick** [reply]
I know this is an old post, but I finally figured out how to set the default printer

[main]
default_printer: printer name

[printer printer name]
